src/pages/index.js

```diff
@@ -0,0 +1,319 @@
+    closeModal(evt.target);
+  }
+}
+
+function handleEscKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_is-opened");
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+const openModal = (modal) => {
+  modal.classList.add("modal_is-opened");
+  document.addEventListener("keydown", handleEscKey);
+  modal.addEventListener("click", handleOverlayClick);
+};
+
+const closeModal = (modal) => {
+  modal.classList.remove("modal_is-opened");
+  document.removeEventListener("keydown", handleEscKey);
+  modal.removeEventListener("click", handleOverlayClick);
+};
+
+const addOpenModalListener = (trigger, modal, cb) => {
+  trigger.addEventListener("click", () => {
+    if (cb) cb();
+    openModal(modal);
+  });
+};
+
+const addCloseModalListener = (button, modal) => {
+  button.addEventListener("click", () => closeModal(modal));
+};
+
+function handleLikeClick(likeBtn, cardId) {
+  const isLiked = likeBtn.classList.contains("card__like-btn_active");
+  api
+    .changeLikeStatus(cardId, !isLiked)
+    .then((card) => {
+      const isNowLiked = card.isLiked;
+      likeBtn.classList.toggle("card__like-btn_active", isNowLiked);
+    })
+    .catch((err) => console.error("Error changing like status:", err));
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.cloneNode(true);
+  cardElement.querySelector(".card__title").textContent = data.name;
+  const cardImage = cardElement.querySelector(".card__image");
+  cardImage.alt = data.name;
+  cardImage.src = data.link;
+  const likeBtn = cardElement.querySelector(".card__like-btn");
+  if (data.isLiked) {
+    likeBtn.classList.add("card__like-btn_active");
+  } else {
+    likeBtn.classList.remove("card__like-btn_active");
+  }
+  likeBtn.addEventListener("click", () => handleLikeClick(likeBtn, data._id));
+  cardElement
+    .querySelector(".card__del-btn")
+    .addEventListener("click", () => handleDelCard(cardElement, data));
+  cardImage.addEventListener("click", () => openPreviewModal(data));
+  return cardElement;
+}
+
+function renderCard(data, container) {
+  const cardElement = getCardElement(data);
+  container.prepend(cardElement);
+}
+
+function openPreviewModal(data) {
+  previewImage.src = data.link;
+  previewImage.alt = data.name;
+  previewCaption.textContent = data.name;
+  openModal(previewModal);
+}
+
+function handleDelCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  submitDeletePostBtn.disabled = false;
+  submitDeletePostBtn.classList.remove("modal__submit_inactive");
+  openModal(deletePostModal);
+}
+
+function handleDelSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(submitDeletePostBtn, "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deletePostModal);
+      selectedCard = null;
+      selectedCardId = null;
+      disableBtn(submitDeletePostBtn, settings);
```

Comment:
` disableBtn(submitDeletePostBtn, settings);` is not needed here. You don't need to disable the confirmation submit button in `.then`. It should always be enabled